src/lib.rs

```diff
@@ -2,26 +2,48 @@ pub mod env;
 pub mod stats;
 pub mod store;
 
-use crate::stats::CpuUsageValue;
+use crate::stats::{
+    CgroupVersion, CpuUsageValue, detect_cgroup_version, get_cgroup_v1_mount_points,
+    get_cgroup_v2_mount_point,
+};
 use crate::store::StatsEntry;
 use std::thread;
 use tracing::{debug, error};
 
 pub fn run_acolyte() {
     let stat_interval = env::get_stat_interval();
     loop {
+        let cgroup_detect = detect_cgroup_version("/proc/self/cgroup").ok();
+        let v2_mount_point = match cgroup_detect {
+            Some(CgroupVersion::V2) => get_cgroup_v2_mount_point("/proc/mounts").ok(),
+            _ => None,
+        };
+        let v1_mount_points = match cgroup_detect {
+            Some(CgroupVersion::V1) => get_cgroup_v1_mount_points("/proc/mounts").ok(),
+            _ => None,
+        };
```

Comment:
I... don't think this needs to be done in the loop? We probably shouldn't expect cgroup versions to change during Acolyte's lifetime?

Reply:
nope, shouldn't, can be outside of the loop